src/org/zaproxy/zap/extension/websocket/WebSocketProxy.java

```diff
@@ -1001,4 +1012,52 @@ public boolean isAllowAPI() {
 	public void setAllowAPI(boolean allowAPI) {
 		this.allowAPI = allowAPI;
 	}
+	
+	/**
+	 * Return an instance of {@link HandshakeConfig} which is used to establish a new WebSocket Connection
+	 * see {@link ServerConnectionEstablisher#send(HandshakeConfig)} and {@link WebSocketProxy#reEstablishConnection(HandshakeConfig)}
+	 * The {@link HandshakeConfig} includes the HttpHandshake of the current connection. In addition, the method adds the
+	 * {@link WebSocketObserver} and {@link WebSocketSenderListener}  of the current connection to HandshakeConfig
+	 *
+	 * @return the HandshakeConfig
+	 * @throws IllegalArgumentException if an error occurred while trying to retrieve the HTTP handshake from history
+	 */
+	public HandshakeConfig getHandShakeConfig() throws IllegalArgumentException {
+		HandshakeConfig handshakeConfig;
+		try {
+			handshakeConfig = new HandshakeConfig(new HttpMessage(handshakeReference.getHttpMessage().getRequestHeader()));
+			for(WebSocketObserver webSocketObserver : observerList){
+				handshakeConfig.addChannelObserver(webSocketObserver);
+			}
+			
+			for(WebSocketSenderListener senderListener : senderListenerList ){
+				handshakeConfig.addChannelSenderListener(senderListener);
+			}
+		} catch (Exception e) {
+			throw new IllegalArgumentException("An error occurred while trying to retrieve the HTTP handshake from history ", e);
+		}
+		return handshakeConfig;
+	}
+	
+	/**
+	 * Re-establish the current connection. Re-establishing by creating a new connection with
+	 * the existing Handshake.
+	 * {@link ServerConnectionEstablisher#send(HandshakeConfig)}
+	 *
+	 * @param handshakeConfig the handshake config
+	 * @return if everything goes well, will return the new instance of {@link WebSocketProxy}
+	 * @throws IOException if an I/O error occurred
+	 * @throws RequestOutOfScopeException if url it's out of score. That happened when ZAP runs to safe/protected mode
```

Comment:
There's a typo in scope. Just noticed that `WebSocketPanelSender` also needs to be updated (same for the `IOException`).